lib/segment/src/index/field_index/geo_index.rs

```diff
@@ -587,6 +594,32 @@ impl PayloadFieldIndex for GeoMapIndex {
             return Some(estimation);
         }
 
+        if let Some(geo_polygon) = &condition.geo_polygon {
+            let (exterior_hashes, interior_hashes) =
+                polygon_hashes_estimation(geo_polygon, GEO_QUERY_MAX_REGION);
+            // The polygon cardinality estimation should consider its exterior and interiors.
+            // Therefore, we compute exterior estimation first and then subtract all interior estimation.
+            let mut exterior_estimation = self.match_cardinality(&exterior_hashes);
+
+            for interior in &interior_hashes {
+                let interior_estimation = self.match_cardinality(interior);
+                exterior_estimation.min = max(0, exterior_estimation.min - interior_estimation.max);
+                exterior_estimation.max = max(
+                    exterior_estimation.min,
+                    exterior_estimation.max - interior_estimation.min,
+                );
+                exterior_estimation.exp = max(
+                    exterior_estimation.exp - interior_estimation.exp,
+                    exterior_estimation.min,
```

Comment:
I am always confused my computation of the expected cardinality.

Why do you `exterior_estimation.min` here?

Reply:
since we also needs to ensure the exp is in the range of [min, max]. This logic also apply to `match_cardinality` https://github.com/qdrant/qdrant/blob/a6164ccda89b00eb361686ed40b90da9fbb1b2c3/lib/segment/src/index/field_index/geo_index.rs#L309